action.yml

```diff
@@ -0,0 +1,114 @@
+    description: 'Whether to run drivers tests. Default disabled.'
+    required: false
+    default: 'false'  
+  static:
+    description: 'Whether to run Falco in static mode during tests. Default disabled.'
+    required: false
+    default: 'false'
+  show-all:
+    description: 'Whether to upload all tests summary, not just failed.'
+    required: false
+    default: 'false'
+  sudo:
+    description: 'Specify a sudo command. Put it empty when sudo is not available.'
+    required: false
+    default: 'sudo'
+
+outputs:
+  report:
+    description: "Generated report xml"
+    value: ${{ steps.store-outputs.outputs.report }} 
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Go
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: "${{ github.action_path }}/go.mod"
+
+    # note: this converts the output of go test into a junit-compatible,
+    # which can later be processed by test-summary/action to upload
+    # a Markdown report on the GitHub Actions workflow.
+    - name: Install go-junit-report
+      shell: bash
+      run: |
+        go install github.com/jstemmer/go-junit-report/v2@latest
+
+    - name: Generate test files
+      working-directory: ${{ github.action_path }}
+      shell: bash
+      run: |
+        go generate ./...
+        
+    - name: Install needed artifacts using falcoctl
+      if: ${{ inputs.static == 'false' }}
+      shell: bash
+      run: |
+        ${{ inputs.sudo }} mkdir -p /usr/share/falco/plugins
+        ${{ inputs.sudo }} falcoctl artifact install k8saudit-rules
+        ${{ inputs.sudo }} falcoctl artifact install cloudtrail-rules
+   
+    - name: Install dependencies for falco-driver-loader tests
+      if: ${{ inputs.arch == 'x86_64' }}
+      shell: bash
+      run: |
+        ${{ inputs.sudo }} apt update -y
+        ${{ inputs.sudo }} apt install -y --no-install-recommends build-essential clang make llvm gcc dkms linux-headers-$(uname -r)
+    
+    - name: Install dependencies for falco-driver-loader tests
+      if: ${{ inputs.arch == 'aarch64' }}
+      uses: self-actuated/get-kernel-sources@master
+    
+    - name: Run tests
+      working-directory: ${{ github.action_path }}
+      env:
+        # fixme(leogr): this is a workaround for https://github.com/falcosecurity/falco/issues/2784
+        HOST_ROOT: ""
+      shell: bash  
+      run: |
+        if ${{ inputs.test-falco == 'true' }}; then
+          ./build/falco.test -falco-static=${{ inputs.static }} -test.timeout=90s -test.v >> ./report.txt 2>&1 || true
+        fi
+        if ${{ inputs.static == 'false' }}; then
+          if ${{ inputs.test-falcoctl == 'true' }}; then
+            ./build/falcoctl.test -test.timeout=90s -test.v >> ./report.txt 2>&1 || true
+          fi
+          if ${{ inputs.test-k8saudit == 'true' }}; then
+            ./build/k8saudit.test -test.timeout=90s -test.v >> ./report.txt 2>&1 || true
+          fi
+          if ${{ inputs.test-drivers == 'true' }}; then
```

Comment:
IIRC we have some issues on arm64... are we sure we want to run these tests also on arm64 when `test-drivers=true`?

Reply:
Yep for now i wanted to try that again; after all, we can revert if anything fails.
But we can skip them entirely on arm64 too.

Reply:
After all it is just a matter of disabling `drivers test` on arm64 by the caller.
In this repo i'd try to stay as abstract as possible.

Reply:
What i mean is that the caller (Falco CI) will do:
```
- uses: falcosecurity/testing@hash
  with:
    test-drivers: ${{ inputs.arch == x86_64 && 'true' || 'false' }}
```

Reply:
yep makes sense